app/src/test/java/org/greenstand/android/TreeTracker/orgpicker/OrgPickerViewModelTest.kt

```diff
@@ -0,0 +1,68 @@
+package org.greenstand.android.TreeTracker.orgpicker
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.runBlockingTest
+import kotlinx.coroutines.test.setMain
+import org.greenstand.android.TreeTracker.MainCoroutineRule
+import org.greenstand.android.TreeTracker.models.organization.Destination
+import org.greenstand.android.TreeTracker.models.organization.Org
+import org.greenstand.android.TreeTracker.models.organization.OrgRepo
+import org.greenstand.android.TreeTracker.utils.getOrAwaitValueTest
+import org.junit.After
+import org.junit.Assert.*
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.mockito.Mockito
+
+@ExperimentalCoroutinesApi
+class OrgPickerViewModelTest {
+    /*
+     Here the Rule runs coroutines on the main thread
+     */
+
+    @get:Rule
+    val instantTaskExecutorRule = InstantTaskExecutorRule()
+
+    @get:Rule
+    val mainCoroutineRule = MainCoroutineRule()
+
+    private val orgRepo = Mockito.mock(OrgRepo::class.java)
+
+    private lateinit var orgPickerViewModel: OrgPickerViewModel
+
+    /*
+     Setting the main dispatcher to Dispatcher.Unconfined and
+     initializing the treeImageReviewViewModel object.
+     */
+
+    @Before
+    fun setUp() {
+        Dispatchers.setMain(Dispatchers.Unconfined)
+        orgPickerViewModel = OrgPickerViewModel(orgRepo)
+    }
+
+    /*
+     Reset the main dispatcher after each test.
+     */
+
+    @After
+    fun tearDown() {
+        Dispatchers.resetMain()
+    }
+
+    /*
+     Test if setOrg sets to the current org to null.
+     */
+
+    @Test
+    fun `test for set org`() = runBlockingTest {
+        val org = null
+        org?.let { orgPickerViewModel.setOrg(it) }
+        val actualState = orgPickerViewModel.state.getOrAwaitValueTest()
+        assertEquals(org, actualState.currentOrg)
+    }
```

Comment:
This test does not actually test anything.

To test the org viewmodel we should:
- ensure it loads the proper current org returned from the orgRepo, as well as the list all orgs. The current org should always be in the list of orgs
- verify the orgRepo gets a new org set when we use the viewmodel to set a new org

Reply:
I am having a problem with this test specifically while implementing it as you have requested. I'll ignore this test for now as I continue to see how i'll go about it. Or if there is anyone willing to help it will be better :)